### DISCUSSION TITLE
Is WRITE_ALL_MASTERS for index changes required in 3.1

### DISCUSSION THREAD

**Discussion by timw** [orientechnologies/orientdb#9799] [General] [answered]
Hi there

We have a distributed production system where sometimes one of the nodes will become unstable, and cause the entire database system to become unavailable for writes.

We've traced this to the use of the `WRITE_ALL_MASTERS` quorum level for any transactions that involve unique index changes (which is the majority of our write workload) rather than the conventional `WRITE` level, which would succeed since sufficient nodes are still available.

In 3.2 the use of `WRITE_ALL_MASTERS` was removed in cec84e1b920b68cb9529b76152ea5c3177111f1f, but I can't see any obvious other changes to transaction handling that allowed that change.

Could this fix be applied to 3.1, or am I missing something that makes it safe only in 3.2?

**Comment by tglman** [comment, marked as answer]
Hi,

Yes this is one of the major changes in 3.2.x version, being the OrientDB distributed protocol a multi-master with quorum based transaction confirmation, to allow multiple transaction happen on the same time, it need something to check if a node is at the same recent state of the data included in the transactions, for do this OrientDB use the document version, that cover most of the cases, except for unique indexes, in 3.2.x has been included also a version for each unique index key, this allow to have a quorum check of the transaction, in <3.1.x this version is missing so is needed a all nodes participating at the transaction with a unique index need to confirm the transaction, not just a quorum.

Regards
